tests/target/chains-no-overflow.rs

```diff
@@ -9,20 +9,16 @@ fn main() {
                        .ddddddddddddddddddddddddddd
                        .eeeeeeee();
 
-    x().y(|| {
-           match cond() {
-               true => (),
-               false => (),
-           }
+    x().y(|| match cond() {
```

Comment:
I'm not sure this is an improvement readability wise. Could we go with the "fixed_broken_closure" solution of #863?